pytorch_lightning/strategies/launchers/spawn.py

```diff
@@ -0,0 +1,191 @@
+        # transfer back the best path to the trainer
+        if trainer.checkpoint_callback:
+            trainer.checkpoint_callback.best_model_path = str(spawn_output.best_model_path)
+
+        # TODO: pass also best score
+        # load last weights
+        if spawn_output.weights_path is not None:
+            ckpt = self._strategy.checkpoint_io.load_checkpoint(spawn_output.weights_path)
+            trainer.lightning_module.load_state_dict(ckpt)  # type: ignore[arg-type]
+            self._strategy.checkpoint_io.remove_checkpoint(spawn_output.weights_path)
+
+        trainer.state = spawn_output.trainer_state
+
+        # get the `callback_metrics` and set it to the trainer
+        if is_overridden("get_from_queue", trainer.lightning_module):
+            # only in case the user does not override it.
+            # TODO: Remove the if in v1.7
+            trainer.lightning_module.get_from_queue(spawn_output.extra)
+        self.get_from_queue(trainer, spawn_output.extra)
+
+    def _collect_rank_zero_results(self, trainer: "pl.Trainer", results: Any) -> Optional["_SpawnOutput"]:
+        rank_zero_debug("Finalizing the DDP spawn environment.")
+        checkpoint_callback = trainer.checkpoint_callback
+        best_model_path = checkpoint_callback.best_model_path if checkpoint_callback else None
+
+        # requires to compute the state_dict on all processes in case Metrics are present
+        state_dict = trainer.lightning_module.state_dict()
+
+        if self._strategy.global_rank != 0:
+            return None
+
+        # save the last weights
+        weights_path = None
+        if trainer.state.fn == TrainerFn.FITTING:
+            weights_path = os.path.join(trainer.default_root_dir, ".temp.ckpt")
+            self._strategy.checkpoint_io.save_checkpoint(state_dict, weights_path)
+
+        # adds the `callback_metrics` to the queue
+        extra = _FakeQueue()
+        if is_overridden("add_to_queue", trainer.lightning_module):
+            # TODO: Remove the if in v1.7
+            trainer.lightning_module.add_to_queue(extra)
+        self.add_to_queue(trainer, extra)
+
+        return _SpawnOutput(best_model_path, weights_path, trainer.state, results, extra)
+
+    def add_to_queue(self, trainer: "pl.Trainer", queue: "_FakeQueue") -> None:
+        """Appends the :attr:`trainer.callback_metrics` dictionary to the given queue. To avoid issues with memory
+        sharing, we cast the data to numpy.
+
+        Args:
+            trainer: reference to the Trainer.
+            queue: the instance of the queue to append the data.
+        """
+        callback_metrics: dict = apply_to_collection(
+            trainer.callback_metrics, torch.Tensor, lambda x: x.cpu().numpy()
+        )  # send as numpy to avoid issues with memory sharing
+        queue.put(callback_metrics)
+
+    def get_from_queue(self, trainer: "pl.Trainer", queue: "_FakeQueue") -> None:
```

Comment:
in lightning_module.py we have deprecation messages that still reference the DDPSpawnStrategy.add_to_queue and get_from_queue. We need to update these mentions